standard/lexical-structure.md

```diff
@@ -615,33 +615,86 @@ The type of a *boolean_literal* is `bool`.
 
 #### 6.4.5.3 Integer literals
 
-Integer literals are used to write values of types `int`, `uint`, `long`, and `ulong`. Integer literals have two possible forms: decimal and hexadecimal.
+Integer literals are used to write values of types `int`, `uint`, `long`, and `ulong`. Integer literals have three possible forms: decimal and, hexadecimal, and binary.
 
 ```ANTLR
 Integer_Literal
     : Decimal_Integer_Literal
     | Hexadecimal_Integer_Literal
+    | Binary_Integer_Literal
     ;
 
 fragment Decimal_Integer_Literal
-    : Decimal_Digit+ Integer_Type_Suffix?
+    : Decimal_Digits Integer_Type_Suffix?
+    ;
+
+fragment Decimal_Digits
+    : Decimal_Digit
+    | Decimal_Digit Decimal_Digits_And_Underscores? Decimal_Digit
```

Comment:
It feels like it would be nice to avoid having to have `Decimal_Digits_And_Underscores` - we effectively need "zero or more Decimal_Digit_And_Underscore, in a non-greedy way". @Nigel-Ecma is my niggle warranted, or is this just "what we do"?

Reply:
@jskeet – your niggle is warranted. Seeing `A : B?; B : C+` rather than `A : C*` hurts one’s eyes ;-) Here is a grammar keeping in the style of the original:

```ANTLR
Integer_Literal
    : Decimal_Integer_Literal
    | Hexadecimal_Integer_Literal
    | Binary_Integer_Literal
    ;

fragment Decimal_Integer_Literal
    : Decimal_Digit Decorated_Decimal_Digit* Integer_Type_Suffix?
    ;

fragment Decorated_Decimal_Digit:
    : '_'* Decimal_Digit
    ;
       
fragment Decimal_Digit
    : '0'..'9'
    ;
    
fragment Integer_Type_Suffix
    : 'U' | 'u' | 'L' | 'l' |
      'UL' | 'Ul' | 'uL' | 'ul' | 'LU' | 'Lu' | 'lU' | 'lu'
    ;
    
fragment Hexadecimal_Integer_Literal
    : ('0x' | '0X') Decorated_Hex_Digit+ Integer_Type_Suffix?
    ;

fragment Decorated_Hex_Digit:
    : '_'* Hex_Digit
    ;
       
fragment Hex_Digit
    : '0'..'9' | 'A'..'F' | 'a'..'f'
    ;
   
fragment Binary_Integer_Literal
    : ('0b' | '0B') Decorated_Binary_Digit+ Integer_Type_Suffix?
    ;

fragment Decorated_Binary_Digit:
    : '_'* Binary_Digit
    ;
       
fragment Binary_Digit
    : '0' | '1'
    ;
```

The `Decorated_x` productions could be inlined, but I think the balance of readability vs. conciseness favours keeping them.

Reply:
Bike-shedding on names:

- `Seperated_Digit`
- `Separable_Digit`

Will stick with Decorated for now, and consider other options for final review.